promtimer/annotations.py

```diff
@@ -0,0 +1,181 @@
+# limitations under the License.
+#
+
+from os import path
+import time
+import json
+from dateutil import parser as dateparser
+import urllib.request
+
+ANNOTS_URL = 'http://localhost:13300/api/annotations'
+ANNOTS_HEADERS = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+}
+ANNOTS_EVENTS_START = {
+    'rebalance_start': 'rebalance',
+    'failover_start': 'failover',
+}
+ANNOTS_EVENTS_END = {
+    'rebalance_finish': 'rebalance',
+    'failover_finish': 'failover',
+}
+ANNOTS_EVENT_TAGS = {
+    # 'dataset_created': 'success',
+    # 'dataset_dropped': 'warning',
+
+    'analytics_index_created': 'success',
+    'analytics_index_dropped': 'warning',
+
+    # 'task_finished': 'success',
+    # 'task_started': 'info',
+
+    'backup_removed': 'warning',
+    'backup_paused': 'warning',
+    'backup_resumed': 'info',
+    'backup_plan_created': 'success',
+    'backup_plan_deleted': 'warning',
+    'backup_repo_created': 'success',
+    'backup_repo_deleted': 'warning',
+    'backup_repo_imported': 'success',
+    'backup_repo_archived': 'success',
+
+    'rebalance_start': 'info',
+    'rebalance_finish': 'info',
+
+    'failover_start': 'warning',
+    'failover_end': 'warning',
+
+    'node_joined': 'success',
+    'node_went_down': 'warning',
+
+    # 'eventing_function_deployed': 'success',
+    # 'eventing_function_undeployed': 'warning',
+
+    'fts_index_created': 'success',
+    'fts_index_dropped': 'warning',
+
+    'index_created': 'success',
+    'index_deleted': 'warning',
+    # 'indexer_active': 'info',
+    # 'index_built': 'info',
+
+    'bucket_created': 'success',
+    'bucket_deleted': 'warning',
+    'bucket_updated': 'info',
+    'bucket_flushed': 'warning',
+
+    # 'dropped_ticks': 'info',
+    'data_lost': 'failure',
+    'server_error': 'failure',
+    'sigkill_error': 'failure',
+    'lost_connection_to_server': 'failure',
+
+    # 'LDAP_settings_modified': 'info',
+    # 'password_policy_changed': 'info',
+    # 'group_added': 'success',
+    # 'group_deleted': 'warning',
+    # 'user_added': 'success',
+    # 'user_deleted': 'warning',
+
+    'XDCR_replication_create_started': 'info',
+    'XDCR_replication_remove_started': 'info',
+    'XDCR_replication_create_failed': 'failure',
+    'XDCR_replication_create_successful': 'success',
+    'XDCR_replication_created': 'success',
+    'XDCR_replication_remove_failed': 'failure',
+    'XDCR_replication_remove_successful': 'success',
+}
+
+def retry_get(url, retries):
+    req = urllib.request.Request(url=url, data=None)
+    success = False
+    get = None
+    while (not success) and (retries > 0):
+        try:
+            get = urllib.request.urlopen(req).read()
+            success = True
+        except:
+            print('Attempting connection to Grafana, retrying...',retries,'retries left')
+            retries -= 1
+            time.sleep(0.5)
+    return get
+
+def create_annotations():
```

Comment:
This function is too long. Should be broken up into multiple functions. E.g. once you know you have a events.log file, the routine to parse that and create the annotations in Grafana should be its own function.